runtime/src/stakes/serde_stakes.rs

```diff
@@ -2,13 +2,74 @@ use {
     super::{StakeAccount, Stakes, StakesEnum},
     crate::stake_history::StakeHistory,
     im::HashMap as ImHashMap,
-    serde::{ser::SerializeMap, Serialize, Serializer},
+    serde::{ser::SerializeMap, Deserialize, Deserializer, Serialize, Serializer},
     solana_sdk::{clock::Epoch, pubkey::Pubkey, stake::state::Delegation},
     solana_stake_program::stake_state::Stake,
     solana_vote::vote_account::VoteAccounts,
     std::sync::Arc,
 };
 
+/// Wrapper struct with custom serialization to support serializing
+/// `Stakes<StakeAccount>` as `Stakes<Stake>` without doing a full deep clone of
+/// the stake data. Serialization works by building a `Stakes<&Stake>` map which
+/// borrows `&Stake` from `StakeAccount` entries in `Stakes<StakeAccount>`. Note
+/// that `Stakes<&Stake>` still copies `Pubkey` keys so the `Stakes<&Stake>`
+/// data structure still allocates a fair amount of memory but the memory only
```

Comment:
Where are the pubkeys copied? Isn't Stakes<T> a bunch of nested Arcs + im::HashMap?